src/security/acl_api.c

```diff
@@ -544,6 +544,50 @@ daos_acl_dump(struct daos_acl *acl)
 
 }
 
+static bool
+validate_aces(struct daos_acl *acl)
+{
+	struct daos_ace *current;
+
+	current = daos_acl_get_next_ace(acl, NULL);
+	while (current != NULL) {
+		if (!daos_ace_is_valid(current)) {
+			return false;
+		}
+
+		current = daos_acl_get_next_ace(acl, current);
+	}
+
+	return true;
+}
+
+bool
+daos_acl_is_valid(struct daos_acl *acl)
+{
+	if (acl == NULL) {
+		return false;
+	}
+
+	if (acl->dal_ver != DAOS_ACL_VERSION) {
+		return false;
+	}
+
+	if (acl->dal_len > 0 && acl->dal_len < sizeof(struct daos_ace)) {
+		return false;
+	}
```

Comment:
Shall we also check that the dal_len is not too big?

Reply:
Wondering what the max value should be. Is there a maximum number of ACEs we will support per pool? Or a value that is just too long regardless of how many ACEs? For example suppose there is only one ACE with an unreasonably long principal string.

Should we add a size limit on the ACE principal string as well? Though I'm not sure what the limit would be there either, since we will be supporting "user@domain" format and possibly other identifiers in the future.

Reply:
After chatting with @dpquigl about it, I think we can assume "user@[domain]" identifiers. Maybe 256 bytes (max strlen == 255) is a good limit there?

Reply:
I didn't see a limit in the Amazon documentation but for Google Cloud Platform they have a max of 100 ACEs per bucket. Since a container is our bucket maybe we can say a max of 100 per container but I'm not sure how that would map to the pool level (probably the same number).

Reply:
My point was that dal_len should at least be smaller than the buffer size.
As for an actual identifier limit, I think that it would be reasonable to impose one.

Reply:
I'm not sure how we can determine the real buffer size at this point, though. Would sizeof actually work here? The size of the struct is dynamically allocated in one large chunk, which includes the buffer.